File: gpio/button.py
#read a button
import RPi.GPIO as GPIO

while True:
	GPIO.setmode(GPIO.BOARD) # use the pin number on board
	GPIO.setup(12,GPIO.IN) # pin12 will receive signal 
	input_value = GPIO.input(12) #get the value of pin12
	if input_value == 0: # if the button has been pressed
		print("The button has been pressed.")
		while input_value == 0: #  when the button is pressed, keep check it status until it is released
			input_value = GPIO.input(12)
		GPIO.cleanup()

File: gpio/button.c
// read a button
#include <wiringPi.h>
#include <stdio.h>

#define btn_pin 1 // aka pin 12 BOARD	

int main(void){
	wiringPiSetup(); // initialize wiring
	if (wiringPiSetup()==-1){  //when initialize wiring failed, print messa$
                printf("setup writingPi failed!");
                return 1;
        }

	pinMode(btn_pin,INPUT);//  btn_pin will receive signal 
	//pullUpDnControl (btn_pin,PUD_DOWN); //does not need it when using resistor
	while(1){
		int input_value = digitalRead(btn_pin);// get the  value of btn_pin
		if (input_value == 0){// if the button has been pressed

			printf("The button has been pressed.\n");
			while (input_value == 0){//  when the button is pressed, keep check it status until it is released
				input_value = digitalRead(btn_pin);
			}
		}	
	}
}
